### PR TITLE
Feature v0.10.0 server response

### PR DESCRIPTION
**Pull request by 0xdcarns**
- refactored join
- refactored getting server ips from server for MQ connections

### CODE REVIEW

File: logger/logger.go [hunk 1]
```diff
@@ -36,6 +37,9 @@ func ResetLogs() {
 
 // Log - handles adding logs
 func Log(verbosity int, message ...string) {
+	var mu sync.Mutex
```
Comment (afeiszli) [inline logger/logger.go:40]: What is the mutex for?

Comment (0xdcarns) [reply logger/logger.go:40]: The logger is being accessed by 3 different go routines in v0.10.0 and this helps avoid concurrent map writes. Tested and it doesn't affect performance

**Comment by 0xdcarns** [comment]
rebased on latest dev changes

File: logger/logger.go (hunk 1 above)
Comment (0xdcarns) [reply logger/logger.go:40]: has been removed

**Pull request merged**